### PR TITLE
Adds PR template text.

### PR DESCRIPTION
**Pull request by damonkohler**

### CODE REVIEW

**Comment by jihoonl** [comment]
Keeping the PR up-to-date to master is pretty hard when it is kept unreviewed and other breaking PRs are getting continuosly merged. Can we say `Rebase to master before creating PR`?

**Comment by damonkohler** [comment]
The goal here is to streamline our PR reviewing process to avoid exactly that situation. We're currently experiencing growing pains and appreciate your patience.

Our intent is to offload effort to contributors as our current PR load exceeds the capabilities of our resources.

**Comment by jihoonl** [comment]
I agree with you about the maintainer's burden. But, it really discourages me, as a contributor, to maintain a PR when I see more PRs goes in after rebasing to master against 3-4 commits already.

Can you at least leave a comment that indicates an expected review timeline? So that, I can rebase it to master once just before the review.

**Comment by damonkohler** [comment]
Absolutely. This template is only a small part of our strategy to improve turn around time on PRs. While we probably won't be giving out ETAs for reviews, we intend to be able to respond quickly (less than one work day) in a way that will correctly set expectations.

**Comment by jihoonl** [comment]
Ok. thanks for the explanation.

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 1.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Want to contribute? Great! Make sure you've checked all these boxes before creating your PR:
+
+- [ ] Keep your PR rebased to master.
```
Comment (SirVer) [inline .github/PULL_REQUEST_TEMPLATE.md]: Actually I think this should now be removed. With Wally rebasing should only happen when there are merge conflicts.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+- [ ] Keep your PR under 200 lines of code and address a single concern.
+- [ ] Add unit test(s) and documentation (these do not count toward your 200 lines).
+- [ ] Adhere to the [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html).
+- [ ] Run `clang-format -style=Google` on your code (this alone is not enough to ensure you've followed the style guide).
```
Comment (SirVer) [inline .github/PULL_REQUEST_TEMPLATE.md]: I think this can also go away, Wally should run clang-format on merge.

**Comment by cschuet** [comment]
SG send a follow-up

**Pull request merged**